src/ModValue.h

```diff
@@ -178,6 +178,48 @@ struct ModInput
 		}
 		return MIMT_Scalar;
 	}
+	// TODO: Make the field use needs_second to decide whether it needs to call
+	// GetElapsedTimeFromBeat at all. -Kyz
+	bool needs_beat()
+	{
+		switch(m_type)
+		{
+			case MIT_EvalBeat:
+			case MIT_MusicBeat:
+			case MIT_DistBeat:
+			case MIT_StartDistBeat:
+			case MIT_EndDistBeat:
+				return true;
+			default:
+				return false;
+		}
+	}
+	bool needs_second()
+	{
+		switch(m_type)
+		{
+			case MIT_EvalSecond:
+			case MIT_MusicSecond:
+			case MIT_DistSecond:
+			case MIT_StartDistSecond:
+			case MIT_EndDistSecond:
+				return true;
+			default:
+				return false;
+		}
+	}
+	bool needs_y_offset()
+	{
+		switch(m_type)
```

Comment:
If only one case is used for `true`, then wouldn't an `if` statement work better? Or to shorten it even more, perhaps just `return m_type == MIT_YOffset;` instead?